src/test/java/pages/BasePage.java

```diff
@@ -0,0 +1,46 @@
+package pages;
+
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import services.WaitService;
+
+import java.awt.*;
+import java.awt.datatransfer.Clipboard;
+import java.awt.datatransfer.StringSelection;
+
+import static tests.BaseTest.getDriver;
+
+public abstract class BasePage {
+
+    WebDriver driver;
+    WaitService waitService;
+    Actions builder;
+    JavascriptExecutor executor;
```

Comment:
Are you sure it's better to store JavascriptExecutor and Actions as global variables here in BasePage?
Yes, these variables might be used inside all the pageObjects but also they might save the state from previous actions which might cause some errors in the feature... It's netter to create a new instance of these variables every time
Or even it would be better to create JsExecutorUtil and ActionUtil (with constant methods) and create generic methods to work with it and just reuse where it's needed